roles/rhel_subscribe/tasks/main.yml

```diff
@@ -0,0 +1,49 @@
+---
+# TODO: Enhance redhat_subscription module
+#       to make it able to attach to a pool
+#       to make it able to enable repositories
+
+- set_fact:
+    rhel_subscription_user: "{{ lookup('env', 'rhel_subscription_user') | default('None', True) }}"
+    rhel_subscription_pass: "{{ lookup('env', 'rhel_subscription_pass') | default('None', True) }}"
+
+- name: RedHat subscriptions
+  redhat_subscription:
+    username: "{{ rhel_subscription_user }}"
+    password: "{{ rhel_subscription_pass }}"
+    # pool: "^OpenShift Enterprise High Touch Beta$"
+    autosubscribe: yes
+  when: ansible_distribution == "RedHat" and
+        rhel_subscription_user != "None" and
+        rhel_subscription_pass != "None"
+
+- name: Get the OpenShift Enterprise High Touch Beta pool ID
+  shell: "subscription-manager list --available | awk -F : \
+         'BEGIN {s=0}
+          /Subscription Name: *OpenShift Enterprise High Touch Beta/ {s=1}
+          s==1 && /Pool ID:/ {print $2; s=2}'"
+  register: high_touch_beta_pool_id
+  when: deployment_type == "enterprise" and
```

Comment:
Might be better to move all the tasks that depend on deployment_type == enterprise to a new task file (say enterprise.yml), then a conditional include can be used to trigger the enterprise related tasks instead of duplicating when deployment_type == 'enterprise' multiple times.

```
- include enterprise.yml
  when: deployment_type == 'enterprise'
```